libkrun/Makefile

```diff
@@ -1,8 +1,28 @@
 LIBKRUNTIME_CFLAGS =	-Wall -shared -fPIC
 COMMON_CFLAGS = -Wall -pedantic -std=gnu99
+SUDO = sudo
 
-ifeq ("${TRAVIS}", "true")
-	COMMON_CFLAGS += -DTRAVIS
+# Sadly this tool requires root
+VIRT_WHAT = ${SUDO} /usr/sbin/virt-what
+
+# Detect if we are virtualised (Linux only for now XXX)
+ifeq ($(shell uname -s),Linux)
+	VIRTUALISED = 0  # default off
+
+	VIRT_FACTS = $(shell ${VIRT_WHAT})
+	ifneq ("${VIRT_FACTS}","")
+		VIRTUALISED = 1
+	endif
+
+	ifeq ("${TRAVIS}","true")
```

Comment:
Maybe I have missed something very obvious. We check for `TRAVIS` as an env var, we run `virt-what` and that sets `VIRTUALISED = 1/0` but then to find out whether Krun is running on a virtualise system we do this:

```python
def is_virtual(self):
    if os.environ.get("TRAVIS") == "true":
        return True
```

Shouldn't we also be checking for `VIRTUALISED` in that Python code?